plugins/unifying/lu-device.c

```diff
@@ -749,12 +749,40 @@ lu_device_get_guids (LuDevice *device)
 	return priv->guids;
 }
 
+static guint8
+lu_device_get_version_bl_minor (LuDevice *device)
+{
+	const gchar* version_bl = lu_device_get_version_bl (device);
+	guint major, minor, build;
+
+	if (version_bl == NULL)
+		return 0;
+
+	if (sscanf (version_bl, "BOT%u.%u_B%u", &major, &minor, &build) != 3)
+		/* bootloader version is malformed */
+		return 0;
+
+	return minor;
+}
+
 void
 lu_device_add_guid (LuDevice *device, const gchar *guid)
 {
 	LuDevicePrivate *priv = GET_PRIVATE (device);
 	if (!as_utils_guid_is_valid (guid)) {
-		g_ptr_array_add (priv->guids, as_utils_guid_from_string (guid));
+		guint8 bl_version = lu_device_get_version_bl_minor (device);
+		const g_autofree gchar* new_guid = NULL;
+
+		/* Reading the bl version is done here to avoid duplicating code
+		   between runtime and bootloader devices */
+		if (bl_version != 0) {
+			new_guid = g_strdup_printf ("%s:%02d", guid, bl_version);
```

Comment:
So this new GUID is made up of the bootloader version. Is the device REV really the same as the unsigned versions? It seems odd that there's be nothing in the USB descriptor that could tell apart signed from unsigned. Could you upload an lsusb of the signed dongle somewhere please?

Reply:
The LSB of the REV is actually the firmware minor version. So it changes when you update the device. Here's the lsusb: [signed.txt]https://github.com/hughsie/fwupd/files/1486920/signed.txt)
Also's here a diff of signed/unsigned: 
[diff.txt](https://github.com/hughsie/fwupd/files/1487009/diff.txt)